### DISCUSSION TITLE
How can I configure the snapshot file name?

### DISCUSSION THREAD

**Discussion by Indra-db** [mitsuhiko/insta#488] [Q&A]
I've ran into the situation where all my test cases produce the same snapshot file. As a solution I want each test case to decide the snapshot file name.I'm currently using these settings. Is there a way to append the settings to decide which filename to use? 

        insta::with_settings!({filters => vec![
            (r"id: (\d+)\s", "[ID] ")
        ]}, {
            insta::assert_yaml_snapshot!(self.str);
        });

**Comment by Indra-db** [comment]
I came to this solution. 

If there's a better way, do let me know

        let mut settings = insta::Settings::clone_current();
        settings
            ._private_inner_mut()
            .filters((vec![(r"id: (\d+)\s", "[ID] ")]));
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("suffix");
        settings.bind(|| {
            insta::assert_yaml_snapshot!(self.str);
        })

**Reply by mitsuhiko** [reply]
You can make it nicer by hiding it behind a macro. Here is an example from the docs for rstest: https://insta.rs/docs/patterns/#rstest

**Reply by Indra-db** [reply]
is there a way to combine it with the settings macro?

**Comment by max-sixty** [comment]
I'm probably missing something obvious, but why can't we use the standard name parameter here?

(IIUC, this question isn't concerned with the `case` issues from `rstest`...)

```rust
    for name in ["name", "name2", "name3"] {
        assert_snapshot!(name, "Hello World");
    }
```

```
insta review finished
accepted:
  insta/tests/test_basic.rs (name)
  insta/tests/test_basic.rs (name2)
  insta/tests/test_basic.rs (name3)
```
